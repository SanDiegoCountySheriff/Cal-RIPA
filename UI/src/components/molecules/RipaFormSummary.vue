<template>
  <div>
    <div v-if="adminEditing">
      <v-card class="ripa-form-summary mx-auto" :loading="this.loading" flat>
        <v-tabs show-arrows>
          <v-tab>Current State</v-tab>
          <v-tab-item>
            <ripa-form-summary-detail
              :apiStop="apiStop"
              :adminEditing="adminEditing"
              :editButtons="editButtons"
              :onEditAgencyQuestions="onEditAgencyQuestions"
              :onEditStop="onEditStop"
              :onEditPerson="onEditPerson"
            ></ripa-form-summary-detail>
          </v-tab-item>
          <v-tab v-for="stopAudit of this.stopAudits" :key="stopAudit.id">
            {{ getStopAuditDate(stopAudit.id) }}
          </v-tab>
          <v-tab-item v-for="stopAudit of this.stopAudits" :key="stopAudit.id">
            <ripa-form-summary-detail
              :apiStop="stopAudit"
              :adminEditing="adminEditing"
              :title="getStopAuditTitle(stopAudit)"
            ></ripa-form-summary-detail>
          </v-tab-item>
        </v-tabs>
      </v-card>
    </div>
    <div v-else-if="adminViewing">
      <v-card class="ripa-form-summary mx-auto" :loading="this.loading" flat>
        <v-tabs show-arrows>
          <v-tab>Current State</v-tab>
          <v-tab-item>
            <ripa-form-summary-detail
              :apiStop="apiStop"
              :adminViewing="adminViewing"
              title="View Stop"
            ></ripa-form-summary-detail>
          </v-tab-item>
          <v-tab v-for="stopAudit of this.stopAudits" :key="stopAudit.id">
            {{ getStopAuditDate(stopAudit.id) }}
          </v-tab>
          <v-tab-item v-for="stopAudit of this.stopAudits" :key="stopAudit.id">
            <ripa-form-summary-detail
              :apiStop="stopAudit"
              :admin-viewing="adminViewing"
              :title="getStopAuditTitle(stopAudit)"
            ></ripa-form-summary-detail>
          </v-tab-item>
        </v-tabs>
      </v-card>
    </div>
    <div v-else>
      <v-card class="ripa-form-summary mx-auto" :loading="this.loading" flat>
        <ripa-form-summary-detail
          :apiStop="apiStop"
          :adminEditing="adminEditing"
          :editButtons="editButtons"
          :onDeletePerson="onDeletePerson"
          :onCopyPerson="onCopyPerson"
          :onEditAgencyQuestions="onEditAgencyQuestions"
          :onEditStop="onEditStop"
          :onEditPerson="onEditPerson"
        ></ripa-form-summary-detail>
      </v-card>
    </div>
  </div>
</template>

<script>
import RipaFormSummaryDetail from '@/components/molecules/RipaFormSummaryDetail'
import { mapActions } from 'vuex'

export default {
  name: 'ripa-form-summary',

  components: {
    RipaFormSummaryDetail,
  },

  data() {
    return {
      loading: false,
      stopAudits: [],
    }
  },

  created() {
    const submittedStop = JSON.parse(
      localStorage.getItem('ripa_form_submitted_api_stop'),
    )
    if (submittedStop?.isEdited) {
      this.getStopAudits()
    }
  },

  methods: {
    ...mapActions(['getAdminStopAudits']),

    async getStopAudits() {
      this.loading = true
      this.stopAudits = await this.getAdminStopAudits(this.apiStop.id)
      this.stopAudits.sort((a, b) => (a.id < b.id ? 1 : b.id < a.id ? -1 : 0))
      this.loading = false
    },

    getStopAuditDate(stopAuditId) {
      const stopAuditDate = stopAuditId.toString().split('-')[1]
      const parsedDate = new Date(
        Date.UTC(
          stopAuditDate.substring(0, 4),
          stopAuditDate.substring(4, 6) - 1,
          stopAuditDate.substring(6, 8),
          stopAuditDate.substring(8, 10),
          stopAuditDate.substring(10, 12),
        ),
      )
        .toISOString()
        .split('T')[0]
      return parsedDate
    },

    getStopAuditTime(stopAuditId) {
      const stopAuditDate = stopAuditId.toString().split('-')[1]
      const parsedTime = new Date(
        Date.UTC(
          stopAuditDate.substring(0, 4),
          stopAuditDate.substring(4, 6) - 1,
          stopAuditDate.substring(6, 8),
          stopAuditDate.substring(8, 10),
          stopAuditDate.substring(10, 12),
        ),
      ).toLocaleTimeString('en-US')
      return parsedTime
    },

    getStopAuditTitle(stopAudit) {
      return `State before edit on: ${this.getStopAuditDate(
        stopAudit.id,
      )} ${this.getStopAuditTime(stopAudit.id)}`
    },
  },

  props: {
    apiStop: {
      type: Object,
      default: () => {},
    },
    adminEditing: {
      type: Boolean,
      default: false,
    },
    adminViewing: {
      type: Boolean,
      default: false,
    },
    editButtons: {
      type: Boolean,
      default: false,
    },
    onDeletePerson: {
      type: Function,
      default: () => {},
    },
    onCopyPerson: {
      type: Function,
      default: () => {},
    },
    onEditAgencyQuestions: {
      type: Function,
      default: () => {},
    },
    onEditStop: {
      type: Function,
      default: () => {},
    },
    onEditPerson: {
      type: Function,
      default: () => {},
    },
  },
}
</script>

<style lang="scss">
.ripa-form-summary {
  width: 90%;
  border: 1px solid #ccc !important;
}
.v-sheet.v-card {
  border-radius: 4px;
}
.v-window {
  border-radius: 4px;
}
</style>
