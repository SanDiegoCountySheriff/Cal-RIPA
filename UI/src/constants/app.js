export const VERSION = '#{TEMPLATE_VERSION}#'
