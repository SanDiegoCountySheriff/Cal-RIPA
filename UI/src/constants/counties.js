export const COUNTIES = [
  {
    name: 'Alameda',
  },
  {
    name: 'Alpine',
  },
  {
    name: 'Amador',
  },
  {
    name: 'Butte',
  },
  {
    name: 'Calaveras',
  },
  {
    name: 'Colusa',
  },
  {
    name: 'Contra Costa',
  },
  {
    name: 'Del Norte',
  },
  {
    name: 'El Dorado',
  },
  {
    name: 'Fresno',
  },
  {
    name: 'Glenn',
  },
  {
    name: 'Humboldt',
  },
  {
    name: 'Imperial',
  },
  {
    name: 'Inyo',
  },
  {
    name: 'Kern',
  },
  {
    name: 'Kings',
  },
  {
    name: 'Lake',
  },
  {
    name: 'Lassen',
  },
  {
    name: 'Los Angeles',
  },
  {
    name: 'Madera',
  },
  {
    name: 'Marin',
  },
  {
    name: 'Mariposa',
  },
  {
    name: 'Mendocino',
  },
  {
    name: 'Merced',
  },
  {
    name: 'Modoc',
  },
  {
    name: 'Mono',
  },
  {
    name: 'Monterey',
  },
  {
    name: 'Napa',
  },
  {
    name: 'Nevada',
  },
  {
    name: 'Orange',
  },
  {
    name: 'Placer',
  },
  {
    name: 'Plumas',
  },
  {
    name: 'Riverside',
  },
  {
    name: 'Sacramento',
  },
  {
    name: 'San Benito',
  },
  {
    name: 'San Bernardino',
  },
  {
    name: 'San Diego',
  },
  {
    name: 'San Francisco',
  },
  {
    name: 'San Joaquin',
  },
  {
    name: 'San Luis Obispo',
  },
  {
    name: 'San Mateo',
  },
  {
    name: 'Santa Barbara',
  },
  {
    name: 'Santa Clara',
  },
  {
    name: 'Santa Cruz',
  },
  {
    name: 'Shasta',
  },
  {
    name: 'Sierra',
  },
  {
    name: 'Siskiyou',
  },
  {
    name: 'Solano',
  },
  {
    name: 'Sonoma',
  },
  {
    name: 'Stanislaus',
  },
  {
    name: 'Sutter',
  },
  {
    name: 'Tehama',
  },
  {
    name: 'Trinity',
  },
  {
    name: 'Tulare',
  },
  {
    name: 'Tuolumne',
  },
  {
    name: 'Ventura',
  },
  {
    name: 'Yolo',
  },
  {
    name: 'Yuba',
  },
]
