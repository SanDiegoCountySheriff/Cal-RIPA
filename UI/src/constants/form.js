export const OFFICER_ASSIGNMENTS = [
  { name: 'Patrol, traffic enforcement, field operations', value: 1 },
  { name: 'Gang enforcement', value: 2 },
  { name: 'Compliance check', value: 3 },
  { name: 'Special events', value: 4 },
  { name: 'Roadblock or DUI sobriety checkpoint', value: 5 },
  { name: 'Narcotics/vice', value: 6 },
  { name: 'Task force', value: 7 },
  { name: 'K-12 public school', value: 8 },
  { name: 'Investigative/detective', value: 9 },
  { name: 'Other', value: 10 },
]

export const DURATIONS = [
  { name: '0 - 10', value: 1 },
  { name: '11 - 20', value: 2 },
  { name: '21 - 30', value: 3 },
  { name: '31 - 60', value: 4 },
  { name: '60+', value: 5 },
]

export const AGES = [
  { name: '0 - 10', value: 1 },
  { name: '11 - 20', value: 2 },
  { name: '21 - 30', value: 3 },
  { name: '31 - 60', value: 4 },
  { name: '60+', value: 5 },
]

export const DISABILITIES = [
  { name: 'Deafness or difficulty hearing', value: 1 },
  { name: 'Speech impairment or limited use of language', value: 2 },
  { name: 'Blind or limited vision', value: 3 },
  { name: 'Mental health condition', value: 4 },
  {
    name: 'Intellectual or developmental disability, including dementia',
    value: 5,
  },
  { name: 'Other disability', value: 6 },
]

export const GENDERS = [
  { name: 'Male', value: 1 },
  { name: 'Female', value: 2 },
  { name: 'Transgender Male', value: 3 },
  { name: 'Transgender Female', value: 4 },
  { name: 'Gender Noncomforming', value: 5 },
]

export const RACES = [
  { name: 'Asian', value: 1 },
  { name: 'Black/African American', value: 2 },
  { name: 'Hispanic/Latino(a)', value: 3 },
  { name: 'Middle Eastern or South Asian', value: 4 },
  { name: 'Native American', value: 5 },
  { name: 'Pacific Islander', value: 6 },
  { name: 'White', value: 7 },
]

export const STOP_REASONS = [
  { name: 'Traffic Violation', value: 1 },
  { name: 'Reasonable Suspicion', value: 2 },
  {
    name: 'Known to be on Parole / Probation / PRCS / Mandatory Supervision',
    value: 3,
  },
  {
    name: 'Knowledge of outstanding arrest warrant/wanted person',
    value: 4,
  },
  {
    name: 'Investigation to determine whether the person was truant',
    value: 5,
  },
  { name: 'Consensual Encounter resulting in a search', value: 6 },
]

export const TRAFFIC_VIOLATIONS = [
  { name: 'Moving Violation', value: 1 },
  { name: 'Equipment Violation', value: 2 },
  {
    name: 'Non-moving Violation, including Registration Violation',
    value: 3,
  },
]

export const REASONABLE_SUSPICIONS = [
  { name: 'Officer witnessed commission of a crime', value: 1 },
  { name: 'Matched suspect description', value: 2 },
  {
    name: 'Witness or Victim identification of Suspect at the scene',
    value: 3,
  },
  { name: 'Carrying Suspicious Object', value: 4 },
  {
    name: 'Actions indicative of casing a victim or location',
    value: 5,
  },
  { name: 'Suspected of Acting as Lookout', value: 6 },
  { name: 'Actions indicative of drug transaction', value: 7 },
  {
    name: 'Actions indicative of engaging in violent crime',
    value: 8,
  },
  { name: 'Other Reasonable Suspicion of a crime', value: 9 },
]

export const ACTIONS_TAKEN_GENERAL = [
  { name: 'Person removed from vehicle by order', value: 1 },
  { name: 'Person removed from vehicle by physical contact', value: 2 },
  { name: 'Field sobriety test conducted', value: 3 },
  { name: 'Curbside detention', value: 4 },
  { name: 'Handcuffed or flex cuffed', value: 5 },
  { name: 'Patrol car detention', value: 6 },
  { name: 'Canine removed from vehicle or used to search', value: 7 },
  { name: 'Firearm pointed at person', value: 8 },
  { name: 'Firearm discharged or used', value: 9 },
  { name: 'Electronic control device used', value: 10 },
  { name: 'Impact projectile discharged or used', value: 11 },
  { name: 'Canine bit or held person', value: 12 },
  { name: 'Baton or other impact weapon used', value: 13 },
  { name: 'Chemical spray used', value: 14 },
  { name: 'Physical or vehicle contact', value: 15 },
  { name: 'Personal photographed', value: 16 },
  { name: 'Vehicle impounded', value: 22 },
]

export const ACTIONS_TAKEN_SEARCH = [
  { name: 'Asked for consent to search another person', value: 17 },
  { name: 'Search of person was conducted', value: 18 },
  { name: 'Asked for consent to search property', value: 19 },
  { name: 'Search of property was conducted', value: 20 },
]

export const BASIS_FOR_PROPERTY_SEIZURE = [
  { name: 'Safekeeping as allowed by law/statute', value: 1 },
  { name: 'Contraband', value: 2 },
  { name: 'Evidence', value: 3 },
  { name: 'Impound of vehicle', value: 4 },
  { name: 'Impound of vehicle', value: 5 },
]

export const PROPERTY_SEIZED_TYPES = [
  { name: 'Firearm(s)', value: 1 },
  { name: 'Ammunition', value: 2 },
  { name: 'Weapon(s) other than a firearm', value: 3 },
  { name: 'Drugs/narcotics', value: 4 },
  { name: 'Alcohol', value: 5 },
  { name: 'Money', value: 6 },
  { name: 'Drug paraphernalia', value: 7 },
  { name: 'Suspected stolen property', value: 8 },
  { name: 'Cell phone(s) or electronic device(s)', value: 9 },
  { name: 'Vehicle', value: 10 },
  { name: 'Other contraband or evidence', value: 11 },
]
