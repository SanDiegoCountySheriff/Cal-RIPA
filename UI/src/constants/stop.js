export const SUBMISSION_STATUSES = [
  {
    text: 'Failed',
    value: 'Failed',
    editable: true,
  },
  {
    text: 'Submitted',
    value: 'Submitted',
    editable: false,
  },
  {
    text: 'Unsubmitted',
    value: 'Unsubmitted',
    editable: true,
  },
  {
    text: 'Resubmitted',
    value: 'Resubmitted',
    editable: false,
  },
]
