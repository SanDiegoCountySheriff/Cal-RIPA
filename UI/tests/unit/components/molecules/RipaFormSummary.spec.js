import RipaFormSummary from '@/components/molecules/RipaFormSummary.vue'
import { shallowMount, mount } from '@vue/test-utils'
import { API_STOP } from '../../constants/RipaFormContainerTestConstants'
import Vuetify from 'vuetify'
import { computed } from 'vue'

describe('Ripa Form Summary', () => {
  let vuetify
  let wrapper

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const factory = propsData => {
    return shallowMount(RipaFormSummary, {
      vuetify,
      propsData: {
        ...propsData,
        onDeletePerson: jest.fn(),
        onCopyPerson: jest.fn(),
        onEditAgencyQuestions: jest.fn(),
        onEditStop: jest.fn(),
        onEditPerson: jest.fn(),
      },
    })
  }

  it('should match snapshot', () => {
    const apiStop = API_STOP
    wrapper = mount(RipaFormSummary, {
      vuetify,
      propsData: {
        apiStop,
        onDeletePerson: jest.fn(),
        onCopyPerson: jest.fn(),
        onEditAgencyQuestions: jest.fn(),
        onEditStop: jest.fn(),
        onEditPerson: jest.fn(),
      },
      provide: {
        isAdminViewing() {
          return false
        },
        isAdminEditing() {
          return false
        },
        statutes: computed(() => [{ id: 'test' }]),
      },
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
