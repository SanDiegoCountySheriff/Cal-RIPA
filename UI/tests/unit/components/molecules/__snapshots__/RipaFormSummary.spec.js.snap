// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ripa Form Summary should match snapshot 1`] = `
<div>
  <!---->
   
  <div>
    <v-card-stub
      class="ripa-form-summary mx-auto"
      flat="true"
      loaderheight="4"
      tag="div"
    >
      <ripa-form-summary-detail-stub
        oncopyperson="[Function]"
        ondeleteperson="[Function]"
        oneditagencyquestions="[Function]"
        oneditperson="[Function]"
        oneditstop="[Function]"
        title="Review, Edit and Submit"
      />
    </v-card-stub>
  </div>
</div>
`;
